### DISCUSSION TITLE
Preset modes

### DISCUSSION THREAD

**Discussion by traisjames** [kyleneideck/BackgroundMusic#415] [Ideas]
In certain cases I may want all apps muted but 1.  An idea is to have customizable presets that let you quickly mute many apps at once or all but one.

**Comment by kyleneideck** [comment]
Like the solo buttons in DAWs? Yeah, I think that would be useful. And it shouldn't be too hard to implement.

Though, it wouldn't just be a matter of muting all of the other apps automatically. It's still possible for apps to to play audio that doesn't get matched to any of the volume sliders in Background Music. So it would need to mute all audio except for certain apps.

That would still be simple enough to do. It would just mean that in the cases where Background Music doesn't match the app to the right audio (e.g. Chrome sometimes) it would end up muting everything, including the app you clicked the solo button for.

So I think we'd want to have it only enable the solo button for apps that it was able to match to an audio client. That would take a bit more work, but it'd probably be useful for other features as well.
